cucumber/policy/features/support/policy_helpers.rb

```diff
@@ -32,68 +32,127 @@ def invoke status: nil, &block
   end
 
   def load_root_policy policy
-    conjur_api.load_policy("root", policy, method: Conjur::API::POLICY_METHOD_PUT)
+    resource('root').put(policy, header)
   end
 
   def update_root_policy policy
-    conjur_api.load_policy("root", policy, method: Conjur::API::POLICY_METHOD_PATCH)
+    resource('root').patch(policy, header)
   end
 
   def extend_root_policy policy
-    conjur_api.load_policy("root", policy, method: Conjur::API::POLICY_METHOD_POST)
+    resource('root').post(policy, header)
   end
 
-  def load_policy id, policy
-    conjur_api.load_policy(id, policy, method: Conjur::API::POLICY_METHOD_PUT)
+  def load_policy(id, policy)
+    resource(id).put(policy, header)
   end
 
-  def update_policy id, policy
-    conjur_api.load_policy(id, policy, method: Conjur::API::POLICY_METHOD_PATCH)
+  def update_policy(id, policy)
+    resource(id).patch(policy, header)
   end
 
-  def extend_policy id, policy
-    conjur_api.load_policy(id, policy, method: Conjur::API::POLICY_METHOD_POST)
+  def extend_policy(id, policy)
+    resource(id).post(policy, header)
   end
 
-  def make_full_id *tokens
-    super(tokens.join(":"))
+  def create_api_key role
+    login_resource.put(
+      "", Authorization: create_token_header, params: { role: role }
+    )
+  end
+
+  def admin_api_key
+    admin_resource.get
+  end
+
+  def get_login_token login, key
+    RestClient.post(
+      uri('authn', CGI.escape(login), 'authenticate'),
+      key, 'Accept-Encoding': 'Base64'
+    )
   end
 
-  def conjur_api
-    login_as_role('admin', admin_api_key) unless @conjur_api
-    @conjur_api
+  def admin_token
+    RestClient.post(
+      uri('authn', 'admin', 'authenticate'),
+      admin_api_key,
+      'Accept-Encoding': 'Base64'
+    )
+  end
+
+  def admin_resource
+    RestClient::Resource.new(
+     uri('authn', 'login', ''), 'admin', admin_password
+    )
+  end
+
+  def resource id
+    RestClient::Resource.new(
+      uri('policies', 'policy', id)
+    )
+  end
+
+  def login_resource
+    RestClient::Resource.new(
+      uri('authn', 'api_key', ''), 'admin', admin_password
+    )
+  end
+
+  def make_full_id *tokens
+    super(tokens.join(":"))
   end
 
   def json_result
     case @result
     when String
       JSON.parse(@result)
-    when Conjur::PolicyLoadResult
-      JSON.parse(@result.to_json)
     when Hash
       @result
     end
   end
 
-  def admin_api_key
-    @admin_api_key ||= Conjur::API.login('admin', admin_password)
+  def create_token_header token = nil
+    if token.nil?
+      token = admin_token
+    end
+    %Q(Token token="#{token}")
+  end
+
+  def header token = nil
+    { Authorization: create_token_header(token) }
+  end
```

Comment:
Do we need both `header` and `create_token_header`?  It seems like `header` is the only one you'd use at the end of the day?

Also re: naming, similar to my comment above: `create_` is not needed.

Reply:
hmm yeah, the only time I don't use the header function but use "create_token_header"  is when I also use params in the request, which happens I think twice. in those cases I write out both in one hash. though there is probably a simpler way to handle both cases in one function